### PR TITLE
Hidden support

### PR DESCRIPTION
**Pull request by abhinavkgrd**
## Description
- Added Sidebar Option to open Hidden Section after password authentication
- Added hidden section with option same as uncategorized section
- Added support to sync hidden files 
- Added Support to Hide and unhide files.
- Hide option available in All section + collection view 
- Handle downloading multiple default hidden collection

## Test Plan

- Verified Hidden section Sidebar authentication working properly
- Tested delete account flow authentication working properly 
- Tested syncing hidden files from remote
- Tested hiding and unhiding files
- Confirmed hidden files not visible during search/dedupe

### CODE REVIEW

File: apps/photos/src/utils/collection/index.ts [hunk 1]
```diff
@@ -260,12 +272,27 @@ export function isValidMoveTarget(
...
+export function isValidReplacementAlbum(
+    collection: Collection,
+    user: User,
+    wantedCollectionName: string
+) {
+    return (
+        collection.name === wantedCollectionName &&
+        (collection.type === CollectionType.album ||
+            collection.type === CollectionType.folder) &&
+        !isHiddenCollection(collection) &&
```
Comment (ua741) [inline apps/photos/src/utils/collection/index.ts:290]: Why add this check?
Suppose if I have a hidden folder on my machine. I drag it and mark it the folder has hidden. Next time when I am dragging the same folder, we shouldn't be creating new album. Adding a check against `defaultHiddenFolder` and `isIncomingShare` should be good enough.

File: apps/photos/src/services/collectionService.ts [hunk 2]
```diff
@@ -173,24 +179,28 @@ const getCollections = async (
...
-    return getNonHiddenCollections(collections);
+    return includeHidden ? collections : getNonHiddenCollections(collections);
 };
 
 export const getCollectionUpdationTime = async (): Promise<number> =>
     (await localForage.getItem<number>(COLLECTION_UPDATION_TIME)) ?? 0;
 
-export const syncCollections = async () => {
-    const localCollections = await getLocalCollections();
+export const syncCollections = async (includeHidden = false) => {
```
Comment (ua741) [inline apps/photos/src/services/collectionService.ts:193]: What are the cases when we wouldn't want to sync `hiddenCollections`?

File: apps/photos/src/utils/file/index.ts [hunk 3]
```diff
@@ -501,12 +486,12 @@ export const createTypedObjectURL = async (blob: Blob, fileName: string) => {
     return URL.createObjectURL(new Blob([blob], { type: type.mimeType }));
 };
 
-export const getUserOwnedNonTrashedFiles = (files: EnteFile[]) => {
+export const getUserOwnedFiles = (files: EnteFile[]) => {
     const user: User = getData(LS_KEYS.USER);
     if (!user?.id) {
         throw Error('user missing');
     }
-    return files.filter((file) => file.isTrashed || file.ownerID === user.id);
```
Comment (ua741) [inline apps/photos/src/utils/file/index.ts:509]: Can you tell me why was this change done.. How are we now handling trashedFiles?

File: apps/photos/src/utils/collection/index.ts (hunk 1 above)
Comment (abhinavkgrd) [reply apps/photos/src/utils/collection/index.ts:290]: yes yes, it does that only, we currently only have default hidden collection and `isHiddenCollection` checks for it is the defaultHiddenCollection. Will update the function name

**Review by ua741 — DISMISSED** [review]

**Review by ua741 — APPROVED** [review]

**Pull request merged**